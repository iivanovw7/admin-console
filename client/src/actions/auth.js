import * as types from '../constants/ActionTypes';
import Cookies from 'js-cookie';
import * as URL from '../constants/APIurl';
import axios from 'axios';

export const signInAction = ({ email, password }, history) => {

  const inOneWeek = new Date(new Date().getTime() + (1000 * 60 * 60 * 24 * 7));

  return async dispatch => {

    await axios.post(`${URL.PRIVATE_API}/auth/login`, {
      email,
      password
    }, { withCredentials: true })
               .then(response => {

                 Cookies.set('LoggedUserObject', response.data, { expires: inOneWeek });
                 Cookies.set('username', response.data.name, { expires: inOneWeek });

                 dispatch({
                   type: types.AUTHENTICATED
                 });
                 history.push('/statistics');

               })
               .catch(error => {

                 dispatch({
                   type: types.AUTHENTICATION_ERROR
                 });

               });

  };
};

export const signOutAction = history => {

  return async dispatch => {

    await axios.get(`${URL.PRIVATE_API}/auth/logout`, { withCredentials: true })
               .then(response => {
                 Cookies.remove('LoggedUserObject');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               })
               .catch(error => {
                 console.log(error);
                 Cookies.remove('LoggedUserObject');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               });
  };

};


