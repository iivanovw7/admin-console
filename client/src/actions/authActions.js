import * as axios from 'axios';
import Cookies from 'js-cookie';
import * as types from '../constants/actionTypes';
import * as URL from '../constants/api';

export const loginUser = ({ email, password }, history) => {

  //TODO
  // consider new way of authenticating user for shorter period of time, may be use reducer
  // instead of cookies to store data and add special route to check authenticated users after
  // page got rendered
  const inOneWeek = new Date(new Date().getTime() + (1000 * 60 * 60 * 24 * 7));

  return async dispatch => {
    await axios.post(`${URL.PRIVATE_API}/auth/login`, {
      email,
      password
    }, { withCredentials: true })
               .then(response => {
                 Cookies.set('LoggedUserObject', response.data, { expires: inOneWeek });
                 Cookies.set('username', response.data.name, { expires: inOneWeek });
                 Cookies.set('id', response.data._id, { expires: inOneWeek });
                 dispatch({
                   type: types.AUTHENTICATED,
                   payload: response
                 });
                 history.push('/statistics');

               })
               .catch(error => {
                 dispatch({
                   type: types.AUTHENTICATION_ERROR
                 });
               });
  };
};

export const logoutUser = history => {

  return async dispatch => {
    await axios.get(`${URL.PRIVATE_API}/auth/logout`, { withCredentials: true })
               .then(response => {
                 Cookies.remove('LoggedUserObject');
                 Cookies.remove('username');
                 Cookies.remove('id');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               })
               .catch(error => {
                 Cookies.remove('LoggedUserObject');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               });
  };
};
