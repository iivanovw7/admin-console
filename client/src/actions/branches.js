import * as types from '../constants/ActionTypes';
import Cookies from 'js-cookie';
import * as URL from '../constants/APIurl';
import axios from 'axios';

export const getBranches = (page, limit, history) => {

  return async dispatch => {
    await axios({
      method: 'get',
      url: `${URL.PRIVATE_API}/branches`,
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/x-www-form-urlencoded'
      },
      params: {
        page: page || 1,
        limit: limit || 10
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({
          type: types.FETCH_BRANCHES,
          payload: response
        });
      })
      .catch(error => {
        console.log(error);
        Cookies.remove('LoggedUserObject');
        dispatch({
          type: types.UNAUTHENTICATED
        });
        history.push('/');
      });
  };

};

export const addNewBranch = formValues => {

  const { name, email, phone, fax, address, information, status } = formValues;

  return async dispatch => {
    await axios({
      method: 'post',
      url: `${URL.PRIVATE_API}/branches`,
      data: {
        name, email, phone, fax, address, information, status
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({
          type: types.ADD_BRANCH,
          payload: response
        });
      })
      .catch(error => {
        console.log(error);
        dispatch({
          type: types.ERROR
        });
      });

  };


};

export const updateBranch = (formValues, id) => {

  const { name, email, phone, fax, address, information, status } = formValues;

  return async dispatch => {
    await axios({
      method: 'put',
      url: `${URL.PRIVATE_API}/branches/${id}`,
      data: {
        name, email, phone, fax, address, information, status
      },
      withCredentials: true
    })
      .then(response => {
        dispatch({
          type: types.UPDATE_BRANCH,
          payload: response
        });
      })
      .catch(error => {
        console.log(error);
        dispatch({
          type: types.ERROR
        });
      });

  };
};

export const getSingleBranch = (id, history) => {

  return async dispatch => {
    await axios({
      method: 'get',
      url: `${URL.PRIVATE_API}/branches/${id}`,
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/x-www-form-urlencoded'
      },
      withCredentials: true
    })
      .then(response => {
        history.push(`/branches/${id}`);
        dispatch({
          type: types.FETCH_BRANCH,
          payload: response
        });
      })
      .catch(error => {
        console.log(error);
        Cookies.remove('LoggedUserObject');
        dispatch({
          type: types.UNAUTHENTICATED
        });
        history.push('/');
      });
  };

};