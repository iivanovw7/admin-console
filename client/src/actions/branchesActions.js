import * as axios from 'axios';
import Cookies from 'js-cookie';
import * as types from '../constants/actionTypes';
import * as URL from '../constants/api';

export const getBranches = (page, limit, history) => {

  return async dispatch => {
    await axios.get(`${URL.PRIVATE_API}/branches`, {
      params: { page, limit },
      withCredentials: true
    })
               .then(response => {
                 dispatch({
                   type: types.FETCH_BRANCHES,
                   payload: response
                 });
               })
               .catch(error => {
                 console.log(error);
                 Cookies.remove('LoggedUserObject');
                 dispatch({
                   type: types.UNAUTHENTICATED
                 });
                 history.push('/');
               });
  };
};

export const addNewBranch = formValues => {

  return async dispatch => {
    await axios.post(`${URL.PRIVATE_API}/branches`, formValues, { withCredentials: true })
               .then(response => {
                 dispatch({
                   type: types.ADD_BRANCH,
                   payload: response
                 });
               })
               .catch(error => {
                 console.log(error);
                 dispatch({
                   type: types.GENERAL_ERROR
                 });
               });
  };
};

export const updateBranch = (formValues, id) => {

  return async dispatch => {
    await axios.put(`${URL.PRIVATE_API}/branches/${id}`, formValues, { withCredentials: true })
               .then(response => {
                 dispatch({
                   type: types.UPDATE_BRANCH,
                   payload: response
                 });
               })
               .catch(error => {
                 console.log(error);
                 dispatch({
                   type: types.GENERAL_ERROR
                 });
               });
  };
};

export const getSingleBranch = (id, history) => {

  return async dispatch => {
    await axios.get(`${URL.PRIVATE_API}/branches/${id}`, { withCredentials: true })
               .then(response => {
                 history.push(`/branches/${id}`);
                 dispatch({
                   type: types.FETCH_BRANCH,
                   payload: response
                 });
               })
               .catch(error => {
                 console.log(error);
                 dispatch({
                   type: types.FETCH_ERROR
                 });
                 history.push('/branches');
               });
  };
};