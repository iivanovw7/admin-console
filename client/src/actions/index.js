export * from './authActions';
export * from './branchesActions';
export * from './groupsActions';
export * from './rolesActions';
export * from './usersActions';
export * from './ticketsActions';
export * from './messagesActions';
export * from './statisticsActions';