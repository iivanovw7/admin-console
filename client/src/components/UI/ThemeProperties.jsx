import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

