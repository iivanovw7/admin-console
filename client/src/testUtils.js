import checkPropTypes from 'check-prop-types';
import { createStore } from 'redux';
import rootReducer from './reducers';

/**
 * @param {object} initialState - initial state to store
 * @function storeFactory
 * @returns {Store} Redux store
 */
export const storeFactory = initialState => {
  return createStore(rootReducer, initialState);
};

/**
 * Returns node(s) with the given test attributes
 * @param {object} wrapper - shallow wrapper
 * @param {string} val - data-test attribute to find
 * @returns {wrapper}
 */
export const findByTestAttr = (wrapper, val) => {
  return wrapper.find(`[data-test='${val}']`);
};