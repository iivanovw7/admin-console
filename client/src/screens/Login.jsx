import React from 'react';
import { Paper, withStyles, Grid, Button } from '@material-ui/core';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { reduxForm } from 'redux-form';
import { Helmet } from 'react-helmet';
import { loginUser } from '../actions/auth';
import { connect } from 'react-redux';
import { LoginField } from '../components/Login/LoginField';
import { errorMessage } from '../components/UI/Forms/ErrorMessage';
import { validateLogin } from '../components/UI/Forms/validate';
import { LoginFormStyles } from '../components/UI/ThemeProperties';

const Login = props => {

  const { classes, handleSubmit } = props;

  const submit = formValues => {
    props.signInAction(formValues, props.history);
  };

  return (
    <Paper className={classes.loginFormContainer}>
      <form className={classes.formPaper} onSubmit={handleSubmit(submit)}>
        <Typography variant="h5" component="h3">
          Sign In
        </Typography>
        <hr/>
        <LoginField dataType={'email'}/>
        <LoginField dataType={'password'}/>
        <Grid container alignItems="center" justify="space-between">
          <Grid item>
            <Button disableFocusRipple disableRipple style={{ textTransform: 'none' }}
                    variant="text" color="primary">Forgot password ?</Button>
          </Grid>
        </Grid>
        <Grid container justify="flex-start" style={{ marginTop: '10px' }}>
          <Button variant="outlined" color="primary" type="submit"
                  style={{ textTransform: 'none' }}>SIGN IN</Button>
          {errorMessage(props)}
        </Grid>
      </form>
    </Paper>
  );
};

Login.propTypes = {
  classes: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func
};

function mapStateToProps(state) {
  return { errorMessage: state.auth.error };
}

const reduxFormLogin = reduxForm({
  validate: validateLogin,
  form: 'login',
  fields: ['email', 'password']
})(Login);

export default connect(mapStateToProps, { loginUser })(withStyles(LoginFormStyles)(reduxFormLogin));

