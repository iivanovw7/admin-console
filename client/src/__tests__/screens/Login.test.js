import { withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import jest from 'jest-mock';
import React from 'react';
import { connect, Provider } from 'react-redux';
import { reduxForm } from 'redux-form';
import { loginUser } from '../../actions';
import { LoginFormStyles } from '../../components/UI/ThemeProperties';
import Login from '../../screens/Login';
import { storeFactory } from '../../testUtils';

const setup = (initialState = {}) => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  let store = storeFactory(initialState);

  //Splitting HOC components
  const Form = reduxForm({ form: 'login', fields: ['email', 'password'] })(Login);
  const Composition = connect(null, { loginUser })(withStyles(LoginFormStyles, { withTheme: true })(Form));
  const props = {
    email: '',
    password: ''
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders login form with styles, imitates users actions:
   *  click submit button and input fields
   */
  describe('Login screen', () => {
    it('Should render login form with fields and styles', () => {
      const mockFunc = jest.fn();
      wrapper = mount(
        <Provider store={store}>
          <Composition {...props} onSubmit={mockFunc()}/>
        </Provider>
      );
      expect(wrapper.html()).not.toBe(null);
      expect(wrapper.html()).not.toBe(undefined);
      expect(wrapper.length).toEqual(1);

      //checking email field properties
      wrapper.find(Login).at(0).props().value = 'email@test.com';
      expect(wrapper.find(Login).at(0).props().value).toBe('email@test.com');

      //checking email and password field events
      wrapper.find('input').first().simulate('change', { target: { value: 'email@test.com' } });
      expect(wrapper.find('input').first().props().value).toBe('email@test.com');
      wrapper.find('input').last().simulate('change', { target: { value: 'password' } });
      expect(wrapper.find('input').last().props().value).toBe('password');

      //simulating click on submit action
      wrapper.find('button').last().simulate('click');
      expect(mockFunc).toHaveBeenCalled();
      expect(mockFunc).toHaveBeenCalledTimes(1);
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
};

setup();



