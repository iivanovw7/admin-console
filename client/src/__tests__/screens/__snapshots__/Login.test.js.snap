// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login screen Should render login form with fields and styles 1`] = `"<div class=\\"MuiPaper-root-11 MuiPaper-elevation2-15 MuiPaper-rounded-12 ReduxForm-loginFormContainer-6 ReduxForm-loginFormContainer-1\\"><form class=\\"ReduxForm-formPaper-8 ReduxForm-formPaper-3\\"><h3 class=\\"MuiTypography-root-38 MuiTypography-h5-54\\">Sign In</h3><hr><div class=\\"MuiGrid-container-74 MuiGrid-spacing-xs-16-97 MuiGrid-align-items-xs-flex-end-84\\"><div class=\\"MuiGrid-item-75 MuiGrid-grid-xs-true-102 MuiGrid-grid-sm-true-116 MuiGrid-grid-md-true-130\\"><div class=\\"MuiFormControl-root-171 MuiFormControl-fullWidth-174\\"><label class=\\"MuiFormLabel-root-186 MuiFormLabel-filled-190 MuiInputLabel-root-175 MuiInputLabel-formControl-180 MuiInputLabel-animated-183 MuiInputLabel-shrink-182\\" data-shrink=\\"true\\" for=\\"email\\">email</label><div class=\\"MuiInputBase-root-206 MuiInput-root-193 MuiInput-underline-197 MuiInputBase-fullWidth-215 MuiInput-fullWidth-200 MuiInputBase-formControl-207 MuiInput-formControl-194\\"><input aria-invalid=\\"false\\" aria-describedby=\\"email-helper-text\\" class=\\"MuiInputBase-input-216 MuiInput-input-201 MuiInputBase-inputType-219 MuiInput-inputType-204\\" id=\\"email\\" name=\\"email\\" rows=\\"1\\" type=\\"email\\" value=\\"email@test.com\\"></div><p class=\\"MuiFormHelperText-root-223 MuiFormHelperText-filled-229\\" id=\\"email-helper-text\\"> </p></div></div></div><div class=\\"MuiGrid-container-74 MuiGrid-spacing-xs-16-97 MuiGrid-align-items-xs-flex-end-84\\"><div class=\\"MuiGrid-item-75 MuiGrid-grid-xs-true-102 MuiGrid-grid-sm-true-116 MuiGrid-grid-md-true-130\\"><div class=\\"MuiFormControl-root-171 MuiFormControl-fullWidth-174\\"><label class=\\"MuiFormLabel-root-186 MuiFormLabel-filled-190 MuiInputLabel-root-175 MuiInputLabel-formControl-180 MuiInputLabel-animated-183 MuiInputLabel-shrink-182\\" data-shrink=\\"true\\" for=\\"password\\">password</label><div class=\\"MuiInputBase-root-206 MuiInput-root-193 MuiInput-underline-197 MuiInputBase-fullWidth-215 MuiInput-fullWidth-200 MuiInputBase-formControl-207 MuiInput-formControl-194\\"><input aria-invalid=\\"false\\" aria-describedby=\\"password-helper-text\\" class=\\"MuiInputBase-input-216 MuiInput-input-201 MuiInputBase-inputType-219 MuiInput-inputType-204\\" id=\\"password\\" name=\\"password\\" rows=\\"1\\" type=\\"password\\" value=\\"password\\"></div><p class=\\"MuiFormHelperText-root-223 MuiFormHelperText-filled-229\\" id=\\"password-helper-text\\"> </p></div></div></div><div class=\\"MuiGrid-container-74 MuiGrid-align-items-xs-center-82 MuiGrid-justify-xs-space-between-93\\"><div class=\\"MuiGrid-item-75\\"><button class=\\"MuiButtonBase-root-257 MuiButton-root-231 MuiButton-text-233 MuiButton-textPrimary-234 MuiButton-flat-236 MuiButton-flatPrimary-237\\" tabindex=\\"0\\" type=\\"button\\" style=\\"text-transform: none;\\"><span class=\\"MuiButton-label-232\\">Forgot password ?</span></button></div></div><div class=\\"MuiGrid-container-74\\" style=\\"margin-top: 10px;\\"><button class=\\"MuiButtonBase-root-257 MuiButton-root-231 MuiButton-outlined-239 MuiButton-outlinedPrimary-240\\" tabindex=\\"0\\" type=\\"submit\\" style=\\"text-transform: none;\\"><span class=\\"MuiButton-label-232\\">SIGN IN</span><span class=\\"MuiTouchRipple-root-260\\"></span></button></div></form></div>"`;
